### PR TITLE
Seed swap

### PR DESCRIPTION
**Pull request by ebensonm**
Seed swap merge

### CODE REVIEW

File: seed_swap.py [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+        seed_num = randint(1,10000)
+        if (seed_num in used_random_seeds):
+            num_run = num_run
         else:
-            run_pipeline_seed(pipeline, seed, cli_args)
+            num_run += 1
+            run_pipeline_seed(pipeline, seed_num, **cli_args)
 
 def run_pipeline_seed(pipeline, seed, **cli_args):
     pipeline = Pipeline.from_json(string_or_file=pipeline)
```
Comment (e13h) [inline seed_swap.py]: What if you called `from_json_structure()` instead? Then you wouldn't need to even use `json.dumps()`. Is there a reason to convert the pipeline to a string?
```suggestion
    pipeline = Pipeline.from_json_structure(pipeline)
```

File: seed_swap.py [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
 
 logger = logging.getLogger(__name__)
 
 def main(**cli_args):
-    #create the function in the query file that returns the seeds that have already been run on and the pipeline to run on
-    seed_runs, pipeline = get_pipeline_seed(pipeline_id)
+    #get the relevant info from the query                               
+    pipeline, datasets_used, used_random_seeds = get_pipeline("f5478b6c-daad-4693-93c3-f073bae78ddf")
+    #now check if there are no used random seeds
     pipeline = json.dumps(pipeline, indent=4)
```
Comment (e13h) [inline seed_swap.py]: See my other comment
```suggestion
```

File: seed_swap.py (hunk 1 above)
Comment (bjschoenfeld) [reply seed_swap.py]: Yes, when working with pipelines, it should either be a file path or a json-like object.

**Comment by bjschoenfeld** [comment]
All files should be under the `experimenter/` directory.

File: experimenter/run_pipeline.py [hunk 3]
```diff
@@ -137,19 +137,38 @@ def run(self, pipeline: Pipeline, metric_names: list = None) -> list:
...
+                data_params=data_params,
+                metrics=metrics,
+                context=context,
+                volumes_dir=self.volumes_dir,
+                runtime_environment=runtime_environment,
+                random_seed=self.random_seed,
+                scoring_random_seed=self.random_seed, 
+                data_random_seed=self.random_seed
+            )
+
```
Comment (bjschoenfeld) [inline experimenter/run_pipeline.py]: this is a lot of duplicated code. lets set variables based on conditions and then have only one call to `evaluate`.

File: experimenter/execute_pipeline.py [hunk 4]
```diff
@@ -129,11 +129,12 @@ def handle_successful_pipeline_run(
     :param pipeline: the pipeline that was run
     :param score: the results from the execution of the pipeline
     """
-    if score["value"][0] == 0:
-        # F-SCORE was calculated wrong - quit and don't keep this run
-        return
+    #if score["value"][0] == 0:
+    #    # F-SCORE was calculated wrong - quit and don't keep this run
+    #    return
```
Comment (bjschoenfeld) [inline experimenter/execute_pipeline.py]: we should just remove this, not comment it out.

File: experimenter/execute_pipeline.py [hunk 5]
```diff
@@ -129,11 +129,12 @@ def handle_successful_pipeline_run(
...
 
     print_pipeline(pipeline.to_json_structure(), score)
+    logger.warning("Succesfully printed pipeline")
```
Comment (bjschoenfeld) [inline experimenter/execute_pipeline.py]: this is an abuse of a warning. info would probably be better.

**Comment by bjschoenfeld** [comment]
Its up to the author of the `pipeline_parser` branch to decide whether to merge this in and actually merge it. @evanphilipsmith @ebensonm.

The new code also needs some tests.

**Comment by bjschoenfeld** [comment]
@ebensonm I think this PR is now obsolete. If so, can you close it?

**Pull request closed without merging**